### PR TITLE
3.5 - Trait i18n, config cleanup, for BasicPiece, Clone, Delete, Text Label. Configurer changes and tests. Stage 1

### PR DESCRIPTION
**Pull request by BrentEaston**

### CODE REVIEW

**Comment by Cattlesquat** [comment]
It all looks great to me!

**Comment by BrentEaston** [comment]
Conflicts resolved

**Comment by BrentEaston** [comment]
I have managed to screw this up somehow. Should have been patient. Closing this will try and cherry pick changes

File: vassal-app/src/main/java/VASSAL/build/module/properties/GlobalProperty.java [hunk 1]
```diff
@@ -81,7 +81,6 @@ public GlobalProperty(GlobalProperty p) {
   @Override
   public String[] getAttributeDescriptions() {
     return new String[]{
-      Resources.getString("Editor.GlobalProperty.component_type"),
       Resources.getString("Editor.name_label"),
       Resources.getString("Editor.GlobalProperty.initial_value"),
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/build/module/properties/GlobalProperty.java:85]: Extra label inadvertently added.

**Comment by BrentEaston** [comment]
Ok, This is stable and ready for review.  
Framework for rebuilding trait configurer dialogs - Check BasicPiece, Clone, Delete, Text Label, Global Key Command, Area Of Effect, Prototype traits.
Re-worked Auto-Configurers - Check Component configurers such as Map.
Updated String, Int, TranslatingStrinEnum, FormattedString, Color Configurers.

Still to do after this PR is merged:
 - Rest of trait configurers dialogs.
 - Rest of individual Configurers
 - Custom build configurer dialogs.

File: vassal-app/src/main/java/VASSAL/configure/TranslatingStringEnumConfigurer.java [hunk 2]
```diff
@@ -113,6 +121,16 @@ public TranslatingStringEnumConfigurer(String key, String name, String[] validVa
     setValue(initialValue);
   }
 
+  public TranslatingStringEnumConfigurer(String key, String name, String[] validValues, String[] i18nKeys, char initialValue) {
+    this(key, name, validValues, i18nKeys, String.valueOf(initialValue));
+  }
+  public TranslatingStringEnumConfigurer(String[] validValues, String[] i18nKeys, String initialValue) {
+    this (null, null, validValues, i18nKeys, initialValue);
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/configure/TranslatingStringEnumConfigurer.java:128]: Extra space.

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/TranslatingStringEnumConfigurer.java:160]: Tidied up this file and added missing Javadoc as well.

**Comment by uckelman** [comment]
Do the all the `testEquals()` actually need to be that? They look like they properly implement `equals()`, and if so, could just override it.

File: vassal-app/src/main/java/VASSAL/configure/ConfigurerCheckBox.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  public ConfigurerCheckBox(String text) {
+    super(text);
+    adjustMargin();
+  }
+
+  private void adjustMargin() {
+    final Insets insets = getMargin();
+    insets.left = -2;
```
[the remark this answers is not in the document]
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/ConfigurerCheckBox.java:47]: I've check linux and it looks good. Same effect. I'm trying to build a Mac VM to test,

File: vassal-app/src/main/java/VASSAL/configure/NamedHotKeyConfigurer.java [hunk 4]
```diff
@@ -288,4 +194,54 @@ public static String encode(NamedKeyStroke stroke) {
     return s;
   }
 
+  class KeyStrokeAdapter extends KeyAdapter {
```
[the remark this answers is not in the document]
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/NamedHotKeyConfigurer.java:197]: Should just be private. Very specialised, I can't see them ever being over-ridden. I will change that

[the remark this answers is not in the document]
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/NamedHotKeyConfigurer.java:197]: Updated

File: vassal-app/src/main/java/VASSAL/configure/PropertiesWindow.java [hunk 5]
```diff
@@ -22,12 +22,12 @@
...
 
 import javax.swing.Action;
-import javax.swing.Box;
-import javax.swing.BoxLayout;
 import javax.swing.JButton;
 import javax.swing.JDialog;
 
 import VASSAL.i18n.Resources;
+import javax.swing.JPanel;
+import net.miginfocom.swing.MigLayout;
```
[the remark this answers is not in the document]
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/PropertiesWindow.java:29]: Updated

**Comment by BrentEaston** [comment]
testEquals() needs to check specifically that the Class,  Type and the State of two Decorators are the same. Yes, that is a reasonable definition of equals(), however, I am leery of replacing equals(), as Decorators currently default to Object.equals() which just checks a == b. I am concerned that there is code somewhere that depends on this behavior of equals().

**Comment by BrentEaston** [comment]
testEquals() needs to check specifically that the Class,  Type and the State of two Decorators are the same. Yes, that is a reasonable definition of equals(), however, I am leery of replacing equals(), as Decorators currently default to Object.equals() which just checks a == b. I am concerned that there is code somewhere that depends on this behavior of equals().

File: vassal-app/src/main/java/VASSAL/configure/ConfigurerCheckBox.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/configure/ConfigurerCheckBox.java]: Good pick up. The left side of the checkbox is cut off on Mac. The insets are set by the LaF and interact with other settings when the Swing items are created. Playing with insets at a Swing component level is asking for a world of hurt. I have removed this and also the ConfigurerTextField. Doesn't look quite as good on Windows, but still overall better.

**Comment by BrentEaston** [comment]
To test it out, just run this branch, edit a module and open any standard AbstractConfigurable, or edit any piece and check BasicPiece, Clone, Delete, Text Label or Global Key Command.

File: vassal-app/src/main/java/VASSAL/configure/BooleanConfigurer.java [hunk 6]
```diff
@@ -35,6 +39,14 @@ public BooleanConfigurer(String key, String name) {
     this(key, name, Boolean.FALSE);
   }
 
+  public BooleanConfigurer (Boolean val) {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/configure/BooleanConfigurer.java:42]: I'm surprised checkstyle doesn't complain about the space between the name and the `(` here.

**Review by uckelman — APPROVED** [review]

**Pull request merged**